paths.tex

```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
 
 \subsection{Consecutive Shortest Path Distance Algorithm}
 
-\begin{figure}[H]
-\begin{lstlisting}[frame=single]
-  Input : Graph (V,E,T), dist[][][] entries set to Inf
-  Output : dist[][][]
-
-  for (t,src) in {T(e) | e in E } x V:
-    st_e := (src,src)
-    T(e) := (t,t) # or (t, Inf)?
-    R := {st_e}
-    S := {}
-    dist[t][src][src] = 0
-
-    while R != {}:
-      pred_e := pop R
-      for e in edge_neighborhood(pred_e) - union S R:
-        t_int := intersect T(pred_e) T(e)
-        if (t_int != null):
-          dist[t][src][snd e] := min(dist[t][src][fst e] + 1,
-                                        dist[t][src][snd e])
-          insert(e,R)
-      S += pred_e
-    return dist
-\end{lstlisting}
-\caption{An algorithm to compute the lengths of all consecutive temporal shortest
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
+  \ForEach(//loop over nodes in time slices){
+    start node $(t, v_{src})$ in $V \times \mathcal{T}$
+  }{
+    Let $e_{start} = (v_{src}, v_{src})$\;
+    Let shortest interval be $T(e_{start}) = (t, t)$\;
+    Let the reached edges min-heap $R = \{e_{start}\}$\;
+    Let the searched edges $S = \varnothing$\;
+    Let $dist[t][src][src] = 0$\;
+    \While{$R \neq \varnothing$}{
+      Let $e_{pred} = pop(R)$\;
+      \ForEach{$e_{succ} = (u,v) \in N_{e_{pred}}(v) - \{e_{succ}\} - S \cup R$}{
```

Comment:
The notation for the edge neighborhood of a vertex `v` is `N_E(v)`, or, the neighborhood of `v` in the set `E`.  

Cant subtract `\{e_{succ}\}` because we dont know what it is. If you meant `\{e_pred\}` then okay. Ideally we'd really use `peek` instead of `pop` and then on 361 write `Insert pop(R) into S`, or insert `e_pred` into `S` at the beginning of the for-loop. But this is fine as is.